contracts/test/unit/VoltV2.t.sol

```diff
@@ -0,0 +1,384 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {DSTest} from "../unit/utils/DSTest.sol";
+import {Vm} from "../unit/utils/Vm.sol";
+import {VoltV2} from "../../volt/VoltV2.sol";
+import {Core} from "../../core/Core.sol";
+import {getCore, getAddresses, VoltTestAddresses} from "./utils/Fixtures.sol";
+import {ICore} from "../../core/ICore.sol";
+import {stdError} from "../unit/utils/StdLib.sol";
+
+contract UnitTestVoltV2 is DSTest {
+    using SafeCast for *;
+    VoltV2 private volt;
+    ICore private core;
+    VoltTestAddresses public addresses = getAddresses();
+
+    Vm private vm = Vm(HEVM_ADDRESS);
+
+    function setUp() public {
+        core = getCore();
+        volt = new VoltV2(address(core));
+    }
+
+    function testTokenDetails() public {
+        assertEq(volt.name(), "Volt");
+        assertEq(volt.symbol(), "VOLT");
+        assertEq(volt.decimals(), 18);
+        assertEq(volt.totalSupply(), 0);
+    }
+
+    function testDelegate() public {
+        volt.delegate(address(0xFFF));
+        assertEq(volt.delegates(address(this)), address(0xFFF));
+    }
+
+    function testMintSuccessMinter(uint224 voltToMint) public {
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(0xFFF), voltToMint);
+
+        vm.prank(address(0xFFF));
+        volt.delegate(address(0xFFF));
+
+        assertEq(volt.totalSupply(), voltToMint);
+        assertEq(volt.getCurrentVotes(address(0xFFF)), voltToMint);
+        assertEq(volt.balanceOf(address(0xFFF)), voltToMint);
+    }
+
+    function testMintAfterDelegation(uint224 voltToMint) public {
+        vm.assume(voltToMint < type(uint224).max / 2);
+
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(0xFFF), voltToMint);
+
+        vm.prank(address(0xFFF));
+        volt.delegate(address(0xFFF));
+
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(0xFFF), voltToMint);
+
+        assertEq(volt.getCurrentVotes(address(0xFFF)), voltToMint * 2);
+    }
+
+    function testMintFailureUnauthorized() public {
+        vm.startPrank(address(0xFFF));
+        vm.expectRevert("CoreRef: Caller is not a minter");
+        volt.mint(address(0xFFF), 1e18);
+        vm.stopPrank();
+    }
+
+    function testMintFailToVoltContract() public {
+        vm.startPrank(addresses.minterAddress);
+        vm.expectRevert("Volt: cannot transfer to the volt contract");
+        volt.mint(address(volt), 1e18);
+        vm.stopPrank();
+    }
+
+    function testMintFailZeroAddress() public {
+        vm.startPrank(addresses.minterAddress);
+        vm.expectRevert("Volt: cannot transfer to the zero address");
+        volt.mint(address(0), 1e18);
+        vm.stopPrank();
+    }
+
+    function testBurn(uint224 voltToBurn) public {
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(this), voltToBurn);
```

Comment:
assert that after mint and before delegation, voting power is 0 for `address(this)`